internal/util/update/update.go

```diff
@@ -374,7 +374,45 @@ func (u Command) updatePackage(ctx context.Context, subPkgPath, localPath, updat
 
 	// Package deleted from upstream
 	case originExists && localExists && !updatedExists:
-		// Check the diff. If there are local changes, we keep the subpackage.
+		// If the package has been deleted from upstream, we only want to delete
+		// it from local if it has no changes.
+		originUnfetched, err := pkg.IsPackageUnfetched(originPath)
```

Comment:
Why are we doing this check here ? Why can't we make sure that all the 3 sources are fetched when they are initially pulled on to local ?